thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java

```diff
@@ -235,48 +234,57 @@ public Response computeSeverity(@NotNull @QueryParam("collection") String collec
       @QueryParam("from") String fromAddr,
-      @QueryParam("to") String toAddr, @QueryParam("subject") String subject,
+      @QueryParam("to") String toAddr,
       @QueryParam("teHost") String teHost, @QueryParam("smtpHost") String smtpHost,
       @QueryParam("smtpPort") int smtpPort,
       @QueryParam("phantomJsPath") String phantomJsPath) {
 
     // run replay, update function with jobId
     long jobId;
     try {
-      Response response = generateAnomaliesInRange(id, startTimeIso, endTimeIso, isForceBackfill, speedup, false);
+      Response response = generateAnomaliesInRange(id, startTimeIso, endTimeIso, isForceBackfill, speedup, isRemoveAnomaliesInWindow);
       Map<Long, Long> entity = (Map<Long, Long>) response.getEntity();
       jobId = entity.get(id);
     } catch (Exception e) {
       return Response.ok("Failed to start replay!").build();
     }
 
+    long startTime = ISODateTimeFormat.dateTimeParser().parseDateTime(startTimeIso).getMillis();
+    long endTime = ISODateTimeFormat.dateTimeParser().parseDateTime(endTimeIso).getMillis();
     JobStatus jobStatus = detectionJobScheduler.waitForJobDone(jobId);
+    int numReplayedAnomalies = 0;
     if (!jobStatus.equals(JobStatus.COMPLETED)) {
+      //TODO: cleanup done tasks and replay results under this failed job
       return Response.ok("Replay job error with job status: {}" + jobStatus).build();
     } else {
-      LOG.info("Replay completed");
+      numReplayedAnomalies = mergedAnomalyResultDAO.findByStartTimeInRangeAndFunctionId(startTime, endTime, id, false).size();
+      LOG.info("Replay completed with {} anomalies generated.", numReplayedAnomalies);
     }
 
-    // create initial tuning and apply filter
-    Response initialAutotuneResponse = initiateAlertFilterAutoTune(id, startTimeIso, endTimeIso,"AUTOTUNE", 10, "", "");
-    updateAlertFilterToFunctionSpecByAutoTuneId(Long.valueOf(initialAutotuneResponse.getEntity().toString()));
-    LOG.info("Initial alert filter applied");
-
-    // send out email
-    long startTime = ISODateTimeFormat.dateTimeParser().parseDateTime(startTimeIso).getMillis();
-    long endTime = ISODateTimeFormat.dateTimeParser().parseDateTime(endTimeIso).getMillis();
-    emailResource.generateAndSendAlertForFunctions(startTime, endTime, String.valueOf(id), fromAddr, toAddr, subject, true, true, teHost, smtpHost, smtpPort, phantomJsPath);
-    LOG.info("Sent out email");
-    return Response.ok("Replay, Tuning and Notification finished!").build();
-  }
+    if (numReplayedAnomalies > 0) {
+      // create initial tuning and apply filter
+      Response initialAutotuneResponse = initiateAlertFilterAutoTune(id, startTimeIso, endTimeIso,"AUTOTUNE", 10, "", "");
+      updateAlertFilterToFunctionSpecByAutoTuneId(Long.valueOf(initialAutotuneResponse.getEntity().toString()));
+      LOG.info("Initial alert filter applied");
 
+      // send out email
+      String subject = new StringBuilder("Replay results for "
```

Comment:
If there is not anomaly detected, there will be no email send, right? If the purpose of this email is to notify the reply is end, we may want to send empty email with title like 'reply is done, but no anomalies is detected'.